auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/security/authentication/UmsAuthenticationService.java

```diff
@@ -30,8 +31,10 @@ public CloudbreakUser getCloudbreakUser(OAuth2Authentication auth) {
 
     public CloudbreakUser getCloudbreakUser(String userCrn, String principal) {
         String requestId = MDCBuilder.getMdcContextMap().get(LoggerContextKey.REQUEST_ID.toString());
-        Crn crn = Crn.fromString(userCrn);
-        if (crn == null) {
+        Crn crn;
+        try {
+            crn = Crn.safeFromString(userCrn);
```

Comment:
Can we check if userCrn is null and throw UmsAuthenticationException in this case? Instead of catching a NPE.
Other: if userCrn is not null but invalid then it can return null, but you removed crn == null check, was it intended?

Reply:
The code as is should already throw UmsAuthenticationException when userCrn is null. The `Crn.safeFromString` method will throw an NPE in that case, which is caught here and turned into a UmsAuthenticationException.

The removal of `crn == null` here was intentional. The code before used `Crn.fromString`, which can return null when the CRN is not null, but doesn't match the expected CRN pattern. This change switches to `Crn.safeFromString`, which instead throws CrnParseException. That gets caught here and turned into UmsAuthenticationException.

Reply:
I added a unit test to ensure that, when userCrn is null, the code throws UmsAuthenticationException.